index.js

```diff
@@ -1,18 +1,133 @@
-const readline = require('readline');
-const rl = readline.createInterface(process.stdin, process.stdout);
+//share files within project
+const lib = require('./ask');
+
+//choosing which game to play first
+gameChoice();
+async function gameChoice() {
+  let userGameChoice = await lib.ask('Do you want to guess first or should I? \nPress "1" if you want to guess first and press "2" if you want me to guess first. \n')
+  if (parseInt(userGameChoice) === 1) {
+    startHumGuessNum();
+  } else if (parseInt(userGameChoice) === 2) {
+    startCompGuessNum();
+  } else {
+    console.log("Well that's not a choice at all! I guess you don't want to play any of my games :( Goodbye!");
+    process.exit();
+  };
 
-function ask(questionText) {
-  return new Promise((resolve, reject) => {
-    rl.question(questionText, resolve);
-  });
-}
 
-start();
+};
 
-async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
+//computer guesses number
+async function startCompGuessNum() {
+  //range
+  let high = (process.argv[2] || 100);
+  let low = 1;
+
+  console.log(`Let's play a game where you (human) think up a number and I (computer) try to guess it. \nPlease think of a number between 1 and ${high} (inclusive).`);
+  let questionText = "What is your secret number?\nI won't peek, I promise...\n";
+  let secretNumber = await lib.ask(questionText);
   console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+
+  //start of game guess count
+  let guessCount = 0;
+  //computer guess
+  let computerGuess = Math.floor((high - low) / 2) + low;
+  //myAnswer is Y or N
+  let myAnswer = await lib.ask(`Is it ${computerGuess}, yes (y) or no (n)?`);
```

Comment:
There is a point here where the user can break the game.  If I type anything other than "y" or "n", the game doesn't know how to handle the response and will start to hang.  This could be handled in a bunch of different ways, but you could potentially write a function that checks the input, and if it is not an expected input asks the user again, and continues to do so until they enter a valid response.